presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcFileMetadata.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.raptor.storage;
+
+import com.facebook.presto.spi.type.TypeSignature;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public class OrcFileMetadata
+{
+    static final String KEY = "com.facebook.presto.raptor.storage.metadata";
+
+    private final Map<Long, TypeSignature> columnTypes;
+
+    @JsonCreator
+    public OrcFileMetadata(Map<Long, TypeSignature> columnTypes)
```

Comment:
Actually that worked because the method  `public Map<Long, TypeSignature> getColumnTypes()`
was annotated with `@JsonValue` instead `@JsonProperty`.
And as a result whole `OrcFileMetadata` was serialized as unnamed JSON map literal.